plugins/Full_CI_ZMQ/pt2_stoch_routines.irp.f

```diff
@@ -511,15 +514,25 @@ subroutine add_comb(comb, computed, tbc, stbc, ct)
   pt2_weight(1) = psi_coef_generators(1,1)**2
   pt2_cweight(1) = psi_coef_generators(1,1)**2
   
-  do i=2,N_det_generators
+  do i=1,N_det_generators
     pt2_weight(i) = psi_coef_generators(i,1)**2
-    pt2_cweight(i) = pt2_cweight(i-1) + psi_coef_generators(i,1)**2
+  enddo
+
+  ! Important to loop backwards for numerical precision
+  pt2_cweight(N_det_generators) = pt2_weight(N_det_generators)
+  do i=N_det_generators-1,1,-1
+    pt2_cweight(i) = pt2_weight(i) + pt2_cweight(i+1) 
   end do
   
```

Comment:
Just a comment, maybe this can be useful one day.
This is know a `cumulative sum` and can be parallelize [1-2] by a traditional block technique.
[2] = http://arogozhnikov.github.io/2015/12/11/cumulative-sum-as-test-for-your-environment.html
[1] = https://www.pdc.kth.se/education/tutorials/summer-school/programming-exercises-on-openmp/solutions-1/part3a-cumulative-sum